ui/src/app/sensors/components/sensor-details/sensor-details.tsx

```diff
@@ -0,0 +1,100 @@
+import {Context} from '../../../shared/context';
+import {historyUrl} from '../../../shared/history';
+import {services} from '../../../shared/services';
+import {SensorEditor} from '../sensor-editor';
+
+require('../../../workflows/components/workflow-details/workflow-details.scss');
+
+export const SensorDetails = ({match, location, history}: RouteComponentProps<any>) => {
+    // boiler-plate
+    const {navigation, notifications} = useContext(Context);
+    const queryParams = new URLSearchParams(location.search);
+
+    const [namespace] = useState(match.params.namespace);
+    const [name] = useState(match.params.name);
+    const [tab, setTab] = useState<string>(queryParams.get('tab'));
+
+    const [sensor, setSensor] = useState<Sensor>();
+    const [edited, setEdited] = useState(false);
+    const [error, setError] = useState<Error>();
+
+    useEffect(
+        () =>
+            history.push(
+                historyUrl('sensors/{namespace}/{name}', {
+                    namespace,
+                    name,
+                    tab
+                })
+            ),
+        [namespace, name, tab]
+    );
+
+    useEffect(() => {
+        services.sensor
+            .get(name, namespace)
+            .then(setSensor)
+            .then(() => setEdited(false))
+            .then(() => setError(null))
+            .catch(setError);
+    }, [namespace, name]);
+
+    useEffect(() => setEdited(true), [sensor]);
+
+    return (
+        <Page
+            title='Sensor Details'
+            toolbar={{
+                breadcrumbs: [
+                    {title: 'Sensors', path: uiUrl('sensors')},
+                    {title: namespace, path: uiUrl('sensors/' + namespace)},
+                    {title: name, path: uiUrl('sensors/' + namespace + '/' + name)}
+                ],
+                actionMenu: {
+                    items: [
+                        {
+                            title: 'Update',
+                            iconClassName: 'fa fa-save',
+                            disabled: !edited,
+                            action: () =>
+                                services.sensor
+                                    .update(sensor, namespace)
+                                    .then(setSensor)
+                                    .then(() => notifications.show({content: 'Updated', type: NotificationType.Success}))
+                                    .then(() => setEdited(false))
+                                    .then(() => setError(null))
+                                    .catch(setError)
+                        },
+                        {
+                            title: 'Delete',
+                            iconClassName: 'fa fa-trash',
+                            disabled: edited,
+                            action: () => {
+                                if (!confirm('Are you sure you want to delete this sensor object?\nThere is no undo.')) {
```

Comment:
minor - `popup.confirm` is a more aesthetically pleasing dialag

Reply:
done.